crates/forge/src/running.rs

```diff
@@ -229,48 +234,71 @@ fn get_fork_state_reader(
     workspace_root: &Utf8PathBuf,
     fork_targets: &[ForkTarget],
     fork_config: &Option<ForkConfig>,
-) -> Option<ForkStateReader> {
+) -> Result<Option<ForkStateReader>> {
     match &fork_config {
-        Some(ForkConfig::Params(url, block_id)) => Some(ForkStateReader::new(
-            url,
-            *block_id,
-            Some(workspace_root.join(".snfoundry_cache").as_ref()),
-        )),
+        Some(ForkConfig::Params(url, block_id)) => {
+            let block_id = get_block_id(url, *block_id)?;
```

Comment:
I would just name the function `get_latest_block id` and only invoke it `if let BlockIdTag(Latest) = block_id`